### ISSUE TITLE
More metadata and object updates for trials.

### ISSUE THREAD

**Issue by jayunit100** [kubeflow/katib#386]
1. How can we get a list of all the trials along with the objective values? 
- Can't use GetTrials, Get Study .  obj values arent in there
related... 
2. Should we be using the grpc or rest api (i.e are some fields only available from one or the other endpoint type)   ?

**Comment by jayunit100** [comment]
cc @johnugeorge

**Comment by johnugeorge** [comment]
Looks like 'objective_value' field in 'trials' table (https://github.com/kubeflow/katib/blob/master/pkg/db/db_init.go#L38) is not updated.

**Comment by jdplatt** [comment]
Our response looks like this

```{
	"trials": [{
			"trial_id": "a07cf0b203b0f276",
			"study_id": "b87b09ea1318c166",
			"parameter_set": [{
					"name": "--learning_rate",
					"parameter_type": "DOUBLE",
					"value": "0.0179"
				},
				{
					"name": "--num_layers",
					"parameter_type": "INT",
					"value": "1"
				},
				{
					"name": "--num_filters",
					"parameter_type": "INT",
					"value": "30"
				}
			],
			"tags": [{}]
		},
		{
			"trial_id": "b746d6a9a63c96b5",
			"study_id": "b87b09ea1318c166",
			"parameter_set": [{
					"name": "--learning_rate",
					"parameter_type": "DOUBLE",
					"value": "0.0514"
				},
				{
					"name": "--num_layers",
					"parameter_type": "INT",
					"value": "2"
				},
				{
					"name": "--num_filters",
					"parameter_type": "INT",
					"value": "30"
				}
			],
			"tags": [{}]
		}
	]
}```

**Comment by jdplatt** [comment]
One other issue we ran into here when trying to create a plot of how the best value of the metric we are tuning for increases over time was that it's not clear how the trials are ordered. Are they in order of when they were started, when they finished, or some other order? It might be worth adding an extra field to clarify the order the trials were generated in

**Comment by johnugeorge** [comment]
Currently, order is not specified in the getTrials query. https://github.com/kubeflow/katib/blob/master/pkg/db/interface.go#L348

**Comment by jdplatt** [comment]
Ok. I think we need to add this to make it possible to analyze the study and decide when to stop it. I'll raise in the other issue discussing the new API specs.
